File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<Integer> postorder(Node root) {
        if (root == null) return new ArrayList<Integer>();

        List<Integer> ans = new ArrayList<>();

        Stack<Node> trev = new Stack<>();
        Stack<Node> rev = new Stack<>();

        trev.push(root);
        Node t;

        while (!trev.isEmpty()) {
            t = trev.pop();
            rev.push(t);

            for (Node c: t.children) trev.push(c);
        }

        while (!rev.isEmpty()) ans.add(rev.pop().val);

        return ans;
    }
}

File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.kt
/**
 * Definition for a Node.
 * class Node(var `val`: Int) {
 *     var children: List<Node?> = listOf()
 * }
 */

class Solution {
    private val ans: MutableList<Int> = mutableListOf();
    private fun postOrder(root: Node?): Unit {
        if (root == null) return;

        for (c in root.children) postOrder(c);

        ans.add(root.`val`);
    }
    fun postorder(root: Node?): List<Int> {
        if (root == null) return emptyList();
        postOrder(root);
        return ans;
    }
}